### PR TITLE
Make the reconnect action of live more robust

### PR DESCRIPTION
**Pull request by CompuIves**
Now you stay as an owner when you reconnect, and we detect the disconncet state more easily.

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/2928/commits/be8341029f6a153544e7910c19dcbcf265f4a084 is at https://pr2928.build.csb.dev/s/new.

File: packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts [hunk 1]
```diff
@@ -10,21 +9,23 @@ import { Operator } from 'app/overmind';
 import { camelizeKeys } from 'humps';
 import { NotificationStatus } from '@codesandbox/notifications/lib/state';
 
-export const onJoin: Operator<LiveMessage<LiveMessageEvent.JOIN>> = mutate(
-  ({ effects, state }) => {
-    effects.notificationToast.success(
-      state.live.isTeam ? 'Connected to Live Team!' : 'Connected to Live!'
-    );
+export const onJoin: Operator<
+  LiveMessage<{ status: 'connected'; live_user_id: string }>
```
Comment (christianalfoni) [inline packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts]: Why remove the Event type? :)

File: packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts [hunk 2]
```diff
@@ -10,21 +9,23 @@ import { Operator } from 'app/overmind';
...
+> = mutate(({ effects, state }, { data }) => {
+  state.live.liveUserId = data.live_user_id;
```
Comment (christianalfoni) [inline packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts]: IS this not when anyone joins? Or is it when only you join?

File: packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts [hunk 3]
```diff
@@ -398,11 +401,10 @@ export const onConnectionLoss: Operator<LiveMessage> = mutate(
 
 export const onDisconnect: Operator<
   LiveMessage<{
-    reason: string;
+    reason: 'close' | 'inactivity';
```
Comment (christianalfoni) [inline packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts]: Would add a type here as well, like the rest of them :)

A bit confused that you can give it a dictionary type here? I believe this is the "event.type"?

File: packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts (hunk 2 above)
Comment (CompuIves) [reply packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts]: Only when you join. good catch!

File: packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts (hunk 1 above)
Comment (CompuIves) [reply packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts]: I wanted to add payload to the message (`status` and `live_user_id`), is there another way to do this?

Comment (christianalfoni) [reply packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts]: Ah, `LiveMessageEvent.JOIN` was wrong here... sorry!

**Pull request merged**